tools/block-generator/generator/generate.go

```diff
@@ -117,26 +118,33 @@ func MakeGenerator(config GenerationConfig) (Generator, error) {
 		config:                    config,
 		protocol:                  proto,
 		params:                    cconfig.Consensus[proto],
+		genesis:                   bkGenesis,
 		genesisHash:               [32]byte{},
 		genesisID:                 "blockgen-test",
 		prevBlockHash:             "",
 		round:                     0,
-		txnCounter:                0,
+		txnCounter:                startingTxnCounter,
 		timestamp:                 0,
 		rewardsLevel:              0,
 		rewardsResidue:            0,
 		rewardsRate:               0,
 		rewardsRecalculationRound: 0,
 		reportData:                make(map[TxTypeID]TxData),
+		nextdbround:               dbround,
```

Comment:
nit: I'd consider calling this round offset rather than nextdbround since the generator doesn't know about the DB.

Reply:
I'll address this and the other ones in a follow up PR.